### ISSUE TITLE
Remove `pip-tools` support

### ISSUE THREAD

**Issue by charliermarsh** [astral-sh/rye#1185]
`uv` has now been the default for a while and folks have contributed more functionality around it.

**Comment by nazq** [comment]
Would you be able to add me as a collaborator. I already have a fork so am not able to create another

**Comment by charliermarsh** [comment]
Yeah, that's roughly what I was imagining. Can you just create a PR from a non-`main` branch in your existing fork? That would be the easier solution.

**Comment by nazq** [comment]
> Yeah, that's roughly what I was imagining. Can you just create a PR from a non-`main` branch in your existing fork? That would be the easier solution.

I'll try that

**Comment by nazq** [comment]
@charliermarsh here's a first cut, but without any docs updates https://github.com/astral-sh/rye/pull/1226
